src/interface.c

```diff
@@ -289,7 +289,13 @@ CK_RV p11prov_module_init(P11PROV_MODULE *mctx)
     P11PROV_debug("PKCS#11: Initializing the module: %s", mctx->path);
 
     dlerror();
+
+#if P11PROV_ADDRESS_SANITIZER
```

Comment:
Probably better to do something like this in provider.h:
```
#if P11PROV_ADDRESS_SANITIZER
#define P11PROV_DLOPEN_FLAGS RTLD_NOW | RTLD_LOCAL
#else
#define P11PROV_DLOPEN_FLAGS RTLD_NOW | RTLD_LOCAL | RTLD_DEEPBIND
#endif
```

Then this in the code:
`mctx->dlhandle = dlopen(mctx->path, P11PROV_DLOPEN_FLAGS0);`

Reply:
Good point. Initially I thought we might need some more overrides, but it looks lkie there are none needed now.

The third option is to use `LD_PRELOAD` to drop this flag in the function wrapper as we already override the dlclose(). This is something I considered, but found too intrusive, but not touching the code. What do you think?

Reply:
we already have a dlclose() wrapper as you said, adding another one is ok, and keeps the code cleaner.
but either way is fine by me.
Once flags defines are moved to a header the code is also clean enough.

Reply:
Fixed.